### ISSUE TITLE
Scale-down does not work for k8s >=v1.16

### ISSUE THREAD

**Issue by vpnachev** [gardener/autoscaler#46]
**What happened**:
I have created a shoot cluster with k8s of version v1.16
I've deployed some applications that triggered scale-up and new nodes were added, but after I've deleted the applications, scale-down is failing.

Here are the logs of the cluster autoscaler
```bash
I0521 12:12:59.632551       1 cluster.go:81] Fast evaluation: ip-10-222-9-79.eu-west-1.compute.internal for removal
I0521 12:12:59.633637       1 cluster.go:112] Fast evaluation: node ip-10-222-9-79.eu-west-1.compute.internal may be removed
I0521 12:12:59.633649       1 cluster.go:81] Fast evaluation: ip-10-222-28-48.eu-west-1.compute.internal for removal
I0521 12:12:59.633945       1 cluster.go:112] Fast evaluation: node ip-10-222-28-48.eu-west-1.compute.internal may be removed
I0521 12:12:59.633958       1 cluster.go:81] Fast evaluation: ip-10-222-27-232.eu-west-1.compute.internal for removal
I0521 12:12:59.634758       1 cluster.go:112] Fast evaluation: node ip-10-222-27-232.eu-west-1.compute.internal may be removed
I0521 12:12:59.634771       1 cluster.go:81] Fast evaluation: ip-10-222-17-46.eu-west-1.compute.internal for removal
I0521 12:12:59.635342       1 cluster.go:112] Fast evaluation: node ip-10-222-17-46.eu-west-1.compute.internal may be removed
I0521 12:12:59.635353       1 cluster.go:81] Fast evaluation: ip-10-222-5-177.eu-west-1.compute.internal for removal
I0521 12:12:59.635485       1 cluster.go:112] Fast evaluation: node ip-10-222-5-177.eu-west-1.compute.internal may be removed
E0521 12:12:59.880361       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:00.881364       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:01.882436       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:02.883508       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:03.884618       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:04.885786       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:05.886821       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:06.888108       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:07.889141       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:08.890477       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:09.891577       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:10.892875       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:11.893811       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:12.894829       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:13.896076       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
I0521 12:13:14.103233       1 static_autoscaler.go:253] No unschedulable pods
E0521 12:13:14.897351       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:15.898421       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
E0521 12:13:16.899560       1 reflector.go:134] github.com/gardener/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:293: Failed to list *v1beta1.DaemonSet: the server could not find the requested resource
```

I set the flags 
```yaml
--scale-down-unneeded-time=1m0s
--scale-down-delay-after-add=1m10s
```
to speed-up scale down, but this did not help at all.

**What you expected to happen**:
To scale underutilized worker pool.

**How to reproduce it (as minimally and precisely as possible)**:
Create a shoot cluster with k8s version v1.16 or higher with min!=max, deploy and then undeply some workload. 

**Anything else we need to know**:

/kind bug
/priority critical

**Environment**:
The version is `gardener/autoscaler@v0.9.0`

**Comment by prashanth26** [comment]
I ran a few quick tests on the cluster pointed out by @vpnachev . The scale down seems to work. Might have been a false positive. 

Two reasons for this,

1. There were pod anti-affinities on the cluster which were preventing a scale down. 
2. Also there is an initial startup delay in scaledown which prevents CA to scale down in the initial few minutes (4-5mins after startup). Probably one of the configurable flags. MIght need to figure this flag out and initialize it appropriately. 

However, this issue doesn't seem to be affecting clusters to the best of my knowledge.

**Comment by hardikdr** [comment]
@prashanth26 Thanks for verifying.
@vpnachev can you please confirm, if this is really an issue or probably a misinterpretation? 
There are also other flags like `--scale-down-delay-after-failure=1m` you might want to set.

**Comment by vpnachev** [comment]
Thank you for looking into this issue.

Yesterday before to report this issue I tried with 
```yaml
--scale-down-unneeded-time=1m0s
--scale-down-delay-after-add=1m10s
```

but 5 minutes later I was still having 5 machines, while because of the anti affinity I needed only 3.

Anyway, the errors for `v1beta1.DaemonSet` are spamming the logs and it is very hard to look through them.

**Comment by prashanth26** [comment]
> but 5 minutes later I was still having 5 machines, while because of the anti affinity I needed only 3.

Did you try wait for a while longer? Bcoz I am sure it would've scaled down in a few minutes. When the nodes I mentioned scaled-down in 5mins, I had set params to 30s each. 

> Anyway, the errors for v1beta1.DaemonSet are spamming the logs and it is very hard to look through them.

Yes, we need to fix that by rebaseing the autoscaler with upstream / update k8s version. But this task is tedious as autoscaler's vendoring is broken upstream making our fork's vendoring also troublesome. We will fix this in the future when more priority issues are resolved.

Also for now, we have removed the dependency from the k8s versions in the autoscaler code to a large extent. The warning message is from the informer library, which can be ignored for now.
